### ISSUE TITLE
Items in Inventorio slots delete when using the "You're in Grave Danger" mod (Incompatibility)

### ISSUE THREAD

**Issue by SargRReal** [RubixDev/Inventorio#96]
if you have tools items blocks etc in your inventorio slots and die with the You're in Grave Danger mod, the items will delete upon death, and will not return to you when you retrieve the grave. More specifically, when you right-click the grave to put all the items back in the inventory slots before you died. This is fabric, i'm not sure about forge. This uses the latest versions of both mods (youre-in-grave-danger-1.2.11 and [1.18+] Inventorio 1.6.2)

So basically, the mod's aren't compatible with the risk of item loss. i would really love to see this issue fixed

**Comment by Lizard-Of-Oz** [comment]
Ideally, I recommend reaching out to the mod's developers and asking them to implement compatibility with Inventorio (this is like a 6th gravestone mod reported, can't catch them all)

However, in the near future (hopefully in less than a month) I'll make an optional workaround for Inventorio that will spawn a chest with items from Inventorio slots upon death.
Yes, this basically means Inventorio will have its own bootleg gravestone mod disabled by default %)

**Comment by SargRReal** [comment]
> Ideally, I recommend reaching out to the mod's developers and asking them to implement compatibility with Inventorio (this is like a 6th gravestone mod reported, can't catch them all)
> 
> However, in the near future (hopefully in less than a month) I'll make an optional workaround for Inventorio that will spawn a chest with items from Inventorio slots upon death. Yes, this basically means Inventorio will have its own bootleg gravestone mod disabled by default %)

honestly i think this would make a fantastic feature for inventorio! Like the chest to spawn in your place of death with items could even be a toggleable gamerule like "deathChestAlways true/false". the only issue would concern the items that expand into the deep pockets enchantment and side bars in inventory, how would it all fit into a single chest? perhaps it'll spawn a double chest if certain requirements are met so all items are accounted for. but even then, i could see an issue with the total number of slots a player has being up to 75 (with all enchantments accounted for), but doublechests only has 54 slots. In my opinion, i think a special "Death Chest" or "Coffin" could be added with the adequate amount of inventory slots needed to store all the players item's, but will delete upon being broken/opened with all the items either returning to you in their previous slots or dropping on the floor to be picked up to prevent the death abuse of generating these larger storage chests. 


on a more ambitious note, perhaps this "deathChest" mechanic could be implemented as a chestplate enchantment called something like "Grave Storage I-III".
Grave Storage I will spawn a default single chest with items in your main/default 3x9 slots of inventory, but everything else still drops as items as normal, so if you can't make it in time at least your main inventory got saved.
Grave Storage II will spawn a double chest with all the items in your default 3x9 slots, but also will hold your armor slots, tools belt, utility belt, and main hot bar, which would fit into 53 of it's 54 slots. This would still risk your three deep pocket 3x9 slots to despawn if you don't make it in time.
Lastly, Grave Storage III will spawn the special "Death Chest" to hold the entirety of your 75 slot inventory(including your extra 3x9 rows of deep pocket slots.)

With the added enchantments in mind, the "deathChest true/false" will just determine if the functionality of the enchantment works/shows up at all, while "deathChestAlways true/false" gamerule where if enabled, will always spawn a Death Chest regardless of the enchantment (would make the the enchantment itself useless/disabled from attaining in game since it happens by default if enabled) 
In a sense, gamerule 1 enables/disables the feature entirely, while gamerule 2 enables/disables the forceful spawn of a death chest upon death.

this way, the player can config their experience to have it work like a gravestone mod, an extra enchantment with progressing levels, or not have it there at all. sorry if i made it sound complicated, but it's a suggestion i couldn't not share

</html>
</body>

<body>

I
--

**Comment by SargRReal** [comment]
sorry didn't mean to close it lmao

**Comment by Lizard-Of-Oz** [comment]
I feel like dedicated gravestone mods would be inherently better at doing this job than making Inventorio a gravestone mod on the side. The point of a death chest is to provide barebones compatibility in case a dedicated gravestone mod doesn't have an Inventorio integration.

**Comment by RubixDev** [comment]
Looks like YIGD already added Inventorio compat [7 months ago](https://github.com/B1n-ry/Youre-in-grave-danger/commit/8eaebf05544f4e0e90eae1d9ac4c309c4ed55774). Other grave mods will also have to add compat on their end.
